kstatus/wait/wait.go

```diff
@@ -10,28 +10,59 @@ import (
 
 	"github.com/pkg/errors"
 	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/api/meta"
 	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
```

Comment:
Something to keep in mind - 
Depending on apimachinery libs means these packages can never be depended on by code living in the k/k repo, and possibly won't be usable by any 3rd party that already depends on k/k code (api, apimachinery, etc.). 
 They don't follow semver, which breaks go modules.   This has caused pain for the kustomize api and cli, and we've decided to cut all ties.  This package is introducing new ties :)